src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -256,12 +257,12 @@ public static FormulaValue Text(EvalVisitor runner, EvalVisitorContext context,
                     if (formatString != null && hasNumberFmt)
                     {
                         // It's a datetime, formatted as number. Let's convert it to a number value first
-                        var newNumber = Library.DateTimeToNumber(IRContext.NotInSource(FormulaType.Number), new DateTimeValue[] { dateTimeValue });
+                        var newNumber = Library.DateTimeToNumber(runner, context, IRContext.NotInSource(FormulaType.Number), new DateTimeValue[] { dateTimeValue });
                         resultString = newNumber.Value.ToString(formatString ?? "g", culture);
```

Comment:
Should we change this to 'G'? Because ToString() default is 'G'
https://learn.microsoft.com/en-us/dotnet/api/system.datetime.tostring?view=net-7.0

Reply:
Also for the `format == null` below, we should also check if empty